lesson11/index.js

```diff
@@ -0,0 +1,66 @@
+const ul = document.getElementById('js-lists');
+
+const fetchData = async() => {
+    const response = await fetch('https://jsondata.okiba.me/v1/json/KN8s0211027113740');
+    const obj = await response.json();
+    const listData = obj.data;
+
+    displayLoadingImage ();
+
+    await new Promise(resolve => {
+        setTimeout(() => {
+            resolve(listData);
+        }, 3000);
+    });
+    createList(listData);
+}
+
+const getData = async() => {
+    try {
+        return await fetchData();
+    } catch (error) {
+        errorText("エラーが発生しました");
+        return error;
```

Comment:
this error is returned, but not using error object. right?

i think

```
const getData = async() => { displayLoadingImage (); try { return await fetchData(); } catch (error) { return error; } finally { removeLoadingImage (); } }


getData().catch((e) => {
  errorText(`${e}:エラーが発生しました`);
});

Reply:
Thank you for your reviewing!
> this error is returned, but not using error object.

I thought so, and I fixed -> e5c45d0

[Q]
> This way you can use it and divide getData function job.

I divide the function and executed it and it worked.
But is it needs to be divide?

Reply:
It is easy to understand if you deliberately break the URL to be retrieved to try and cause an error.
```js
const fetchData = async () => {
  try {
    const response = await fetch(
      // I'm gonna break this.
      "https://jsondaaaata.okiba.me/v1/json/KN8s0211027113740"
    );
    const obj = await response.json();
    const listData = obj.data;
    return listData;
  } catch (error) {
    //here should be an error with TypeError: Failed to fetch.
    //What you see is TypeError: value is not iterable.
    //Because the error is returned but not used anywhere.
    return error;
  }
};
```
I wrote the same thing in Japanese.
[codesandbox](https://codesandbox.io/s/little-architecture-22ylv?file=/index.js)

Reply:
Thank you for telling me how to verify.
I understand that if I just return an error, the error will not be displayed correctly.
I fixed -> 057b5af

Reply:
Q
error is returned, where is this used?
```js
        errorText(`エラーが発生しました: 詳細 ${error}`);
        return error;
```

Reply:
Thank you for your reviewing!!
I fixed -> d54f3f6